pkg/network/ovn_kubernetes.go

```diff
@@ -1836,3 +1726,191 @@ func isIPSecEnabledInPod(pod v1.PodTemplateSpec, containerName string) bool {
+	}
+	for _, snet := range conf.ServiceNetwork {
+		_, cidr, err := net.ParseCIDR(snet)
+		if err != nil {
+			out = append(out, errors.Wrapf(err, "could not parse spec.serviceNetwork %s", snet))
+			continue
+		}
+		if err := pool.Add(*cidr); err != nil {
+			out = append(out, errors.Errorf("Whole or subset of ServiceNetwork CIDR %s is already in use: %s", snet, err))
+		}
+	}
+
+	oc := conf.DefaultNetwork.OVNKubernetesConfig
+	if oc != nil {
+		// Validate whether join and transit switch CIDRs are from same IP family
+		// as clusterNetwork. Also validate whether provided CIDRs have enough IPs to
+		// allocate to all nodes scheduleable as per cluterNetwork CIDR and hostPrefix.
+		// Note: oc.V4InternalSubnet will be deprecated in future as per k8s guidelines
+		if oc.V4InternalSubnet != "" {
+			if !cnHasIPv4 {
+				out = append(out, errors.Errorf("v4InternalSubnet %s and ClusterNetwork must have matching IP families", oc.V4InternalSubnet))
+			}
+			_, _, err = net.ParseCIDR(oc.V4InternalSubnet)
+			if err != nil {
+				out = append(out, errors.Errorf("v4InternalSubnet is invalid: %s", err))
+			} else if !isV4NodeSubnetLargeEnough(conf.ClusterNetwork, oc.V4InternalSubnet) {
+				out = append(out, errors.Errorf("v4InternalSubnet %s is not large enough for the maximum number of nodes which can be supported by ClusterNetwork", oc.V4InternalSubnet))
+			}
+		}
+		// Note: oc.V6InternalSubnet will be deprecated in future as per k8s guidelines
+		if oc.V6InternalSubnet != "" {
+			if !cnHasIPv6 {
+				out = append(out, errors.Errorf("v6InternalSubnet %s and ClusterNetwork must have matching IP families", oc.V6InternalSubnet))
+			}
+			_, _, err = net.ParseCIDR(oc.V6InternalSubnet)
+			if err != nil {
+				out = append(out, errors.Errorf("v6InternalSubnet is invalid: %s", err))
+			} else if !isV6NodeSubnetLargeEnough(conf.ClusterNetwork, oc.V6InternalSubnet) {
+				out = append(out, errors.Errorf("v6InternalSubnet %s is not large enough for the maximum number of nodes which can be supported by ClusterNetwork", oc.V6InternalSubnet))
+			}
+		}
+
+		if oc.IPv4 != nil {
+			if oc.IPv4.InternalJoinSubnet != "" {
+				if !cnHasIPv4 {
+					out = append(out, errors.Errorf("v4InternalJoinSubnet %s and ClusterNetwork must have matching IP families", oc.IPv4.InternalJoinSubnet))
+				}
+				_, ipv4JoinSwitchCidr, err := net.ParseCIDR(oc.IPv4.InternalJoinSubnet)
+				if err != nil {
+					out = append(out, errors.Errorf("v4InternalJoinSubnet is invalid: %s", err))
+				} else if !isV4NodeSubnetLargeEnough(conf.ClusterNetwork, oc.IPv4.InternalJoinSubnet) {
+					out = append(out, errors.Errorf("v4InternalJoinSubnet %s is not large enough for the maximum number of nodes which can be supported by ClusterNetwork", oc.IPv4.InternalJoinSubnet))
+				}
+				if err := pool.Add(*ipv4JoinSwitchCidr); err != nil {
+					out = append(out, errors.Errorf("Whole or subset of v4InternalJoinSubnet CIDR %s is already in use: %s", oc.IPv4.InternalJoinSubnet, err))
+				}
```

Comment:
Maybe consider to have an adhoc aux function for this section of code that repeats 8 times, like
```
func validateOVNKubernetesSubnets(name, subnet string, otherSubnets *iputil.IPPool, cn []operv1.ClusterNetworkEntry)
```

Reply:
Agreed, I have made changes and pushed. Thanks!

Reply:
Done